### PR TITLE
Stop gitignoring the bin/ folder

### PR DESCRIPTION
**Pull request by viktorsmari**
#### What? Why?
New rails app (`rails new ..`) creates the `bin/` folder and it does not gitignore the `bin/` folder.
When running `rails app:update`, the bin folder also gets updated.

This also allows us to use `bin/rails` to run rails, or `bin/yarn` etc.

**Tip**: 
- Create a shell `alias r=./bin/rails`
- Then we can do `r c` for `bin/rails console`, and `r s` for `bin/rails server`

Using the `bin/` is optional and this PR does not enforce it's use.

Also:
- Rails 6 has the `webpack-dev-server` in the bin folder #7556 
- There is a `bin/setup` - Maybe we move/merge our `script/setup` there? #7624 #7613

Rails 4 introduced the bin folder according to this answer:
https://stackoverflow.com/questions/27584045/whats-inside-my-rails-bin-directory

Official docs:
https://guides.rubyonrails.org/command_line.html
The `app:update` task:
https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task

### CODE REVIEW

**Comment by shen-sat** [comment]
I'd like to add this issue to the epic (https://github.com/openfoodfoundation/openfoodnetwork/issues/7613) if that makes sense @viktorsmari ? First I'll clean up the setup script (https://github.com/openfoodfoundation/openfoodnetwork/issues/7624) and then we can move it into the bin/setup script as per your recommendation

**Comment by viktorsmari** [comment]
@shen-sat Sure if you think it makes sense?
After thinking more about it, maybe we should **not** join the 2 files, and maybe keep both, but reduce the lines in `script/setup`?

All the files in `bin/` are created by rails, and sometimes they get updated on `rails app:update`

I personally like having the files generated by rails **clean**, meaning that I won't put my custom stuff in them. That means every time I run `rails app:update` I don't have to merge my custom stuff, because I keep it in another file. But this is my personal preference, maybe the others want to merge everything.

- The `bin/setup` is creating the database and running `bundle`.
- The `script/setup` is doing more, like copying the `config/application.yml` in place and running custom rake tasks that seed the database.

The only downside to have these `bin/` files **clean** is that we might then have to run 2 scripts, instead of 1 :nerd_face: 

Also the `script/` folder has multiple files, like `script/rails` that won't be needed if this PR is merged.
Then there are more files that have not been touched in 6 years, but we need input from the other devs if we want to remove them. Some of them might still be useful or in use.

File: bin/setup
Comment (sauloperez) [inline bin/setup:36]: Is this bin/setup created by Rails itself?

**Review by sauloperez — COMMENTED** [review]
As a rule of thumb, I tend to agree with @viktorsmari idea of having small and as independent as possible PRs

File: bin/setup
Comment (Matt-Yorkley) [reply bin/setup:36]: I think it's a legacy Rails thing; older versions kept it under `/script` and newer versions keep it under `/bin`. So the placement of the file depends what version of Rails you used to originally initialize your project...

**Review by Matt-Yorkley — COMMENTED** [review]
So before Rails 4 these files lived in `/script` by default and after Rails 4, they live in `/bin` by default, right?

Normally it wouldn't really matter where you stick these files, but there's a bit more complexity here, which is:
- some of these paths related to bin files might be provided slightly different by rbenv wen the gems are updated and the shims are rehashed. I'm not sure what the effects of explicitly providing  `/bin/ruby` and `bin/yarn` here might be in certain contexts..?
- we also might have some code in the `ofn-install` repo that's used for Devops which references some things in the `/scripts` directory, we would need to double check that...

File: bin/setup
Comment (viktorsmari) [reply bin/setup:36]: Yes, everything in this PR is the output of `rails app:update`

**Comment by sauloperez** [comment]
> 
So before Rails 4 these files lived in /script by default and after Rails 4, they live in /bin by default, right?

I think so

**Comment by viktorsmari** [comment]
> Normally it wouldn't really matter where you stick these files, but there might be a bit more complexity here, which is:

Right now this PR is not changing which `rails` we use. It's just adding the `bin` folder created by `rails app:update`. I think another (and more time consuming) PR is needed for deciding when/if to use what. I think it's good to have the default `bin` provided (and maintained) by rails, and our custom scripts separate.

I remember in another project switching over to use `bin/rails` in a Dockerfile, because at some point the image did not find `rails`. Don't remember if it was a problem with bundler inside Docker or rails version updates or something else :man_shrugging: 
https://github.com/fablabbcn/smartcitizen-api/commit/82f909db

> * we also might have some code in the `ofn-install` repo that's used for Devops which references some things in the `/scripts` directory, we would need to double check that pretty carefully...

Definitely, I think another issue / bigger PR could address that.

This PR is a minor step in making each rails upgrade easier, and trying to be closer to rails recommendation.

Also if we plan on using Webpack via the Webpacker gem in the future, it's installation process will add 2 files in the `bin/` folder 

```bash
       ...
       exist    bin
      create    bin/webpack
      create    bin/webpack-dev-server
```

More about the update task:
https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task

**Review by mkllnk — COMMENTED** [review]
This is a great step. After checking out your commit, I have two more scripts in bin:

- rspec
- spring

Do you agree about adding those as well?

**Comment by viktorsmari** [comment]
> Do you agree about adding those as well?

Yes of course :smile: 
I also have those files in other projects. Just curious, how did you generate those files?

**Comment by mkllnk** [comment]
I'm not sure but I guess that I followed https://github.com/rails/spring#setup.
```sh
bundle exec spring binstub --all
```

That also creates rspec and modifies the rails binstubs. I was worried about executing spring in production if we commit this. But if we install the right gemset (without development) then it should be fine:

> On platforms where Spring is installed and supported, this snippet hooks Spring into the execution of commands. In other cases, the snippet will just be silently ignored and the lines after it will be executed as normal.

**Comment by RachL** [comment]
Hello @viktorsmari just checking in to see how you are doing: do you still have time to work on this PR?

**Comment by viktorsmari** [comment]
@RachL I thought this PR would be merged as is, and @mkllnk would add his files in another PR.
I was not 100% sure how @mkllnk added his files :nerd_face:

**Comment by viktorsmari** [comment]
I added the suggestions from @mkllnk  :+1:

**Comment by RachL** [comment]
awesome thanks @viktorsmari I'm moving your PR to code review so it can have a final review before merge.

**Comment by andrewpbrett** [comment]
> we also might have some code in the ofn-install repo that's used for Devops which references some things in the /scripts directory, we would need to double check that pretty carefully...

I think this PR itself wouldn't break anything, since everything in `script/` is still there, but we'd want to update ofn-install to point to `bin/` pretty soon so that we're not maintaining things in two places. 

> I was worried about executing spring in production if we commit this. 

I checked on this, spring doesn't get installed in production, we do `--without development test` in ofn-install when we deploy

> Also if we plan on using Webpack via the Webpacker gem in the future

We now have `webpacker` in the Gemfile if you want to add those two to this PR 👍

**Review by andrewpbrett — APPROVED** [review]

**Comment by viktorsmari** [comment]
@andrewpbrett I think @Matt-Yorkley already added the webpacker binstubs here:

https://github.com/openfoodfoundation/openfoodnetwork/commit/a074f627b266b7c4780062e058ea35d13e7ca9ba

He also stopped gitignoring the `bin` folder, so the binstubs are already on master:
https://github.com/openfoodfoundation/openfoodnetwork/tree/master/bin

**Comment by andrewpbrett** [comment]
@viktorsmari ah I missed that, thanks! We just need a second approval then.

**Review by sauloperez — APPROVED** [review]

**Pull request merged**
